host/windows/syscall.c

```diff
@@ -3032,3 +3032,81 @@ int oe_syscall_nanosleep_ocall(struct oe_timespec* req, struct oe_timespec* rem)
     _set_errno(0);
     return 0;
 }
+
+/*
+**==============================================================================
+**
+** clock_gettime():
+**
+**==============================================================================
+*/
+
+/*
+**==============================================================================
+**
+** POSIX_TO_WINDOWS_EPOCH_TICKS:
+**
+** The ticks elapsed since 1970-01-01 00:00 (UTC). This value was derived with
+** the following function.
+**
+**     LONGLONG get_posix_to_windows_epoch_ticks()
+**     {
+**         SYSTEMTIME st = { .wYear = 1970, .wMonth = 1, .wDay = 1 };
+**         FILETIME ft;
+**         ULARGE_INTEGER x;
+**
+**         SystemTimeToFileTime(&st, &ft);
+**         x.u.LowPart = ft.dwLowDateTime;
+**         x.u.HighPart = ft.dwHighDateTime;
+**
+**         return x.QuadPart;
+**     }
+**
+**==============================================================================
+*/
+
+const LONGLONG POSIX_TO_WINDOWS_EPOCH_TICKS = 0X19DB1DED53E8000;
+
+/* gets the current time and puts it into the buffer pointed to by cur_time */
+static void _clock_gettime_realtime(struct oe_timespec* cur_time)
+{
+    FILETIME ft;
+    ULARGE_INTEGER x;
+    const LONGLONG TICKS_PER_MILLISECOND = 10000UL;
+    const LONGLONG MILLIS_PER_SEC = 1000UL;
+    const LONGLONG NANOS_PER_MILLI = 1000000UL;
```

Comment:
Use names consistent with tests/stdc/enc/enc.cpp , i.e.:
    const uint64_t SEC_TO_USEC = 1000000UL;
    const uint64_t SEC_TO_MSEC = 1000UL;
    const uint64_t MSEC_TO_NSEC = 1000000UL;
Personally I would prefer them both chaned to use PER, and MSEC as in
NSEC_PER_MSEC, MSEC_PER_SEC, etc.  But the point is we should be consistent between the files.